### ISSUE TITLE
joomla_user_agent_rce module error

### ISSUE THREAD

**Issue by astro-train** [rapid7/metasploit-framework#6358]
Greetings! I'am using Kali 2.0 with  4.11.4-2015071403 version of msfconsole. However, when I'am trying to exploit with this module I've got error:
**Exploit failed: NoMethodError undefined method `get_html_meta_elements' for #<Rex::Proto::Http::Response:0x11907098>**
What could it be and how can I solve that issue? Thank you.

**Comment by gmikeska-r7** [comment]
Looks like your framework is out of date, based on your Kali version. Please follow Kali's directions for upgrading and try again. Thanks!

**Comment by astro-train** [comment]
But Kali 2.0 is latest avaliable at the moment, alsot msfconsole version is near the latest (most recent is 4.11.5) also, I've upgraded with apt-get upgrade. So I can't imagine what could it be.

**Comment by FireFart** [comment]
@astro-train the function was added in https://github.com/rapid7/metasploit-framework/commit/12cdd786a68f2642e69a566d488fd1db812af8ca#diff-373b76d959837c360f4fcc157a61bc6c and first released in `4.11.5-2015110801` if we can trust github on this. As your version is `2015071403`, you need to update (maybe using dist-upgrade) because it's too old and does not include the necessary changes.

**Comment by astro-train** [comment]
@FireFart, thanks for replying. I tried to upgrade everything, so everything in the systems seems up to date, but for some reason msfupdate still doesn't work. 

root@kali:~# msfupdate 
[*]
[*] Attempting to update the Metasploit Framework...
[*]

[*] Checking for updates via the APT repository
[*] Note: expect weekly(ish) updates using this method
[*] No updates available

and at the same time:

 root@kali:~# msfconsole -v
Framework Version: 4.11.4-2015071403

I'am confused about this.

**Comment by madmike33** [comment]
@astro-train  give me ur skype i will assist you

**Comment by astro-train** [comment]
@madmike33, here you go: alex.nealex but I don't know if it is a good idea since I'am not native english speaker.

**Comment by busterb** [comment]
From the regular Kali Linux command line, run as root:

```
apt-get update
apt-get dist-upgrade
```

**Comment by astro-train** [comment]
@busterb, hey! The problem is that I've already done those steps for upgrading, even severla times, but still  msfupdate doesn't see any updateds. That's so strange.

**Comment by bcook-r7** [comment]
msfupdate, when run on Kali Linux, just does 'apt-get upgrade' anyway. There's no advantage or anything special about using 'msfupdate', opposed than using Kali's own update mechanism.

If you are not seeing any upgrade packages still, I would recommend seeking help on #kali on freenode, or their support forums.

**Comment by wchen-r7** [comment]
@astro-train Or if you want, you can just git clone metasploit framework from github, that way you don't have to wait for Kali to release the update package.

**Comment by astro-train** [comment]
So I decided to use git. I've cloned msf but when I run msfupdate I got 

An error occurred while installing pg (0.18.3), and Bundler cannot continue.
Make sure that `gem install pg -v '0.18.3'` succeeds before bundling.

And I can't install this pg in any way. Wondering whats wrong with that?

**Comment by busterb** [comment]
if you are now setting up a development environment with git, then you also need to have the development packages, e.g. libpq-dev, installed

https://github.com/rapid7/metasploit-framework/wiki/Setting-Up-a-Metasploit-Development-Environment#install-the-base-dev-packages

**Comment by astro-train** [comment]
@busterb, I run this commaind to install dev packages and got a lot of errors (http://pastebin.com/UFQpzD2B). My sources.list differs, because I use kali 2.0 (using sources from article causing many "404  Not Found" erros from apt-get. When I'am trying to install libpq-dev it says that packet depends on krb5-multidev and comerr-dev but those packages couldn't be updated. Also I'am sorry for being so obtrusive. Maybe I need to add some repos to source.list? I really can't understand why this packages that libpq-dev requires couldn't be installed.

**Comment by busterb** [comment]
It sounds like you do not have your sources.list configured properly, or something is preventing you from accessing Kali's repositories. If you cannot install basic packages, you have a bigger problem than just not having the latest Metasploit.

http://docs.kali.org/general-use/kali-linux-sources-list-repositories

I strongly suggest you seek support from the Kali linux forums or #kali-linux on freenode if putting your sources.list back into their original supported state does not solve the problem.

**Comment by wchen-r7** [comment]
The sources.list documentation is also described here:
https://github.com/rapid7/metasploit-framework/wiki/Setting-Up-a-Metasploit-Development-Environment#update-kali-linux

**Comment by astro-train** [comment]
@busterb , @wchen-r7 much thanks! I've edites sources list (wonde what was wrong with them out of the box?) and then installed all the packets needed. Not I've got this:

/usr/lib/ruby/2.1.0/pathname.rb:48:in `rindex': incompatible character encodings: UTF-8 and ASCII-8BIT (Encoding::CompatibilityError)
	from /usr/lib/ruby/2.1.0/pathname.rb:48:in `chop_basename'
	from /usr/lib/ruby/2.1.0/pathname.rb:244:in `relative?'
	from /usr/lib/ruby/2.1.0/pathname.rb:228:in `absolute?'
	from /usr/lib/ruby/2.1.0/pathname.rb:397:in `block in join'
	from /usr/lib/ruby/2.1.0/pathname.rb:394:in `reverse_each'
	from /usr/lib/ruby/2.1.0/pathname.rb:394:in `join'
	from /root/Рабочий стол/metasploit-framework/lib/metasploit/framework/common_engine.rb:73:in `merge_meterpreter_extensions'
	from /root/Рабочий стол/metasploit-framework/lib/metasploit/framework/common_engine.rb:52:in `block (2 levels) in <module:CommonEngine>'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/initializable.rb:30:in `instance_exec'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/initializable.rb:30:in `run'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /usr/lib/ruby/2.1.0/tsort.rb:226:in `block in tsort_each'
	from /usr/lib/ruby/2.1.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
	from /usr/lib/ruby/2.1.0/tsort.rb:427:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/2.1.0/tsort.rb:347:in `block in each_strongly_connected_component'
	from /usr/lib/ruby/2.1.0/tsort.rb:345:in `each'
	from /usr/lib/ruby/2.1.0/tsort.rb:345:in `call'
	from /usr/lib/ruby/2.1.0/tsort.rb:345:in `each_strongly_connected_component'
	from /usr/lib/ruby/2.1.0/tsort.rb:224:in `tsort_each'
	from /usr/lib/ruby/2.1.0/tsort.rb:205:in `tsort_each'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/initializable.rb:54:in `run_initializers'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/application.rb:215:in `initialize!'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/railtie/configurable.rb:30:in `method_missing'
	from /root/Рабочий стол/metasploit-framework/config/environment.rb:5:in `<top (required)>'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/application.rb:189:in `require'
	from /var/lib/gems/2.1.0/gems/railties-4.0.13/lib/rails/application.rb:189:in `require_environment!'
	from /root/Рабочий стол/metasploit-framework/lib/metasploit/framework/command/base.rb:63:in `require_environment!'
	from /root/Рабочий стол/metasploit-framework/lib/metasploit/framework/command/base.rb:81:in `start'
	from ./msfconsole:48:in `<main>'

Didn't find anything useful on google for this. Is this because I have cyrillic symbols in my path?

**Comment by astro-train** [comment]
Yeah, that was path name issue. It's all working fine now. Thank you, guys, so much for supporting!

**Comment by bcook-r7** [comment]
Sure thing - would you mind updating https://community.rapid7.com/thread/8054 with a link to here and closing that one? Thanks!

**Comment by astro-train** [comment]
@bcook-r7  I didn't find a way to close topic by myself, hope moderators will do this.
